platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamSingleFileRepairIterator.java

```diff
@@ -0,0 +1,196 @@
+                    // update running hash calculator with event.
+                    runningHashCalculator.addObject(dce);
+                } else {
+                    logger.warn(
+                            "Unexpected event in repair iterator, class: {}",
+                            next.getClass().getCanonicalName());
+                }
+            }
+        } catch (final IOException e) {
+            // No error, set next element to null.
+            next = null;
+        }
+    }
+
+    /**
+     * Assumption: that there are no more elements in the event stream iterator. We repair the event stream sequence if
+     * it is missing a final hash by setting the next to the running hash of the previous event returned.
+     */
+    private void attemptRepair() {
+        if (prev.getClassId() == Hash.CLASS_ID) {
+            if (hashCount == 1 && eventCount == 0) {
+                logger.error(EXCEPTION.getMarker(), "Cannot repair, no events in sequence.");
+            }
+        } else if (prev instanceof DetailedConsensusEvent dce) {
+            // event stream did not have a final hash.
+            // repaired by providing the final running hash as the next element in the sequence.
+            try {
+                next = dce.getRunningHash().getFutureHash().getAndRethrow();
+                finalHashComputed = true;
+            } catch (final InterruptedException ex) {
+                logger.error(EXCEPTION.getMarker(), "Interrupted Exception while waiting for running hash.", ex);
+                Thread.currentThread().interrupt();
+            }
+        } else {
+            logger.warn(
+                    "Unexpected event stream pattern in single file repair iterator, "
+                            + "event count is {} and last event class is: {}",
+                    eventCount,
+                    prev.getClass().getCanonicalName());
+        }
+    }
+
+    /**
+     * Sets the next SelfSerializable from the event stream iterator if it exists and updates the running hash. If the
+     * event stream iterator does not have a final hash, the sequence is repaired by setting the next SelfSerializable
+     * to the running hash saved in the previous event.
+     *
+     * @return true if the next event is set, false otherwise.
+     */
+    private boolean findNextWithRepair() {
+        setNextAndUpdateRunningHash();
+        if (next == null && !finalHashComputed) {
+            attemptRepair();
+        }
+        return next != null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public SelfSerializable next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        prev = next;
+        next = null;
+        return prev;
+    }
+
+    /**
+     * Indicates if the event stream has been repaired.
+     *
+     * @return false until and unless the event stream has been repaired, then true is returned.
+     */
+    public boolean wasRepaired() {
```

Comment:
please update the method name and javadoc to match the updated variable, since this class does no actual repairing, just determining if repair is needed and if so, what the final hash is.

Reply:
I agree this does not repair the file, but in my mind it does repair the in-memory expected pattern of `Hash, [1..n] Events, Hash` that the iterator provides.   I will change the name of the method to `finalHashAdded` and the javadoc on the method to reflect this.